src/test/scala/org/ergoplatform/explorer/persistence/repositories/AdProofRepoSpec.scala

```diff
@@ -0,0 +1,35 @@
+package org.ergoplatform.explorer.persistence.repositories
+
+import cats.effect.IO
+import org.ergoplatform.explorer.persistence.{repositories, RealDbTest}
+import org.scalatest.{Matchers, PropSpec}
+import org.scalatestplus.scalacheck.ScalaCheckDrivenPropertyChecks
+
+class AdProofRepoSpec
+  extends PropSpec
+  with Matchers
+  with RealDbTest
+  with ScalaCheckDrivenPropertyChecks {
+
+  import org.ergoplatform.explorer.persistence.models.Generators._
+
+  property("insert/getByHeaderId") {
+    withLiveRepos { (headerRepo, repo) =>
+      forSingleInstance(adProofWithHeaderGen) {
+        case (header, proof) =>
+          headerRepo.insert(header).unsafeRunSync()
+          repo.getByHeaderId(proof.headerId).unsafeRunSync() shouldBe None
+          repo.insert(proof).unsafeRunSync()
+          repo.getByHeaderId(proof.headerId).unsafeRunSync() shouldBe Some(proof)
+      }
+    }
+  }
+
+  private def withLiveRepos(
+    body: (HeaderRepo.Live[IO], AdProofRepo.Live[IO]) => Any
```

Comment:
Can we run tests against abstract Repos?

Reply:
We can, but this would require us to parametrize method or just repeat the signature of `Live` class extension.